### PR TITLE
Problem when materializing multiple clauses, one of them a list (closes #5)

### LINKED ISSUE #5: Problem when materializing multiple clauses, one of them a list
Hi,

There is a bug when we try to materialize a query with multiple clauses and at least one of them is a list. It introduces Scala List inside the JSON, which is incorrect for ElasticSearch. Here are the steps to reproduce.

1. String to parse with at least one clause with a List of possible values.
```scala
"foo = [ \"foo2\" ] and bar = \"bar\""
```
2. When parse the string it produces a correct clause tree for querying.
```scala
Query(CombinedClause(Clause(foo,=,List(foo2)),and,Clause(bar,=,bar)))
```
3. At then moment we try to materialize into the JSON which will be sent to ES, the problem appears
```json
{
    "query": {
        "bool": {
            "filter": [
                {
                    "term": {
                        "foo": {
                            "value": "List(foo2)",
                            "boost": 1
                        }
                    }
                },
                {
                    "term": {
                        "bar": {
                            "value": "bar",
                            "boost": 1
                        }
                    }
                }
            ],
            "disable_coord": false,
            "adjust_pure_negative": true,
            "boost": 1
        }
    },
    "ext": {}
}
```

As you can see, JSON generated try to search for _List(foo2)_ instead of _foo2_, which is the value we wrote.

When there is only one clause and it is a list, it transform correctly.
```scala
"""foo = [ "foo", "bar" ]"""
```
```scala
Query(Clause(foo,=,List(foo, bar)))
```
```json
{
  "query" : {
    "terms" : {
      "foo" : [
        "foo",
        "bar"
      ],
      "boost" : 1.0
    }
  },
  "ext" : { }
}
```

### PR DESCRIPTION
**Pull request by sergiohgz**

### CODE REVIEW

File: lightql-elastic/src/main/scala/com/tecsisa/lightql/mat/elastic/Materializer.scala [hunk 1]
```diff
@@ -30,8 +30,16 @@ object Materializer {
...
+              case EqOp.`=` =>
+                qb.must(nestQuery(c.value match {
+                  case list: List[_] => termsQuery(c.field, list.asJava)
+                  case _             => termQuery(c.field, c.value)
+                }, c.field))
+              case EqOp.!= =>
+                qb.mustNot(nestQuery(c.value match {
+                  case list: List[_] => termsQuery(c.field, list.asJava)
+                  case _             => termQuery(c.field, c.value)
+                }, c.field))
```
Comment (juanjovazquez) [inline lightql-elastic/src/main/scala/com/tecsisa/lightql/mat/elastic/Materializer.scala]: These two cases are identical. Why not?:

```scala
case EqOp.`=` | EqOp.!= => ...
```

File: lightql-elastic/src/test/scala/com/tecsisa/lightql/mat/elastic/SearchSpec.scala [hunk 2]
```diff
@@ -104,6 +104,37 @@ class SearchSpec extends SearchBaseTest {
       val q6 = q("price > 0.99 and stats.rate.stars < 4.5 and date.year >= 2000")
       search("songs") query q6 should haveTotalHits(3)
     }
+
+    "find exact results in queries with multiple clauses and at least one of them have multiple values" in {
```
Comment (juanjovazquez) [inline lightql-elastic/src/test/scala/com/tecsisa/lightql/mat/elastic/SearchSpec.scala]: having multiple values

**Review by juanjovazquez — CHANGES_REQUESTED** [review]
Looking good. Just some minor adjustments. Thanks!.

File: lightql-elastic/src/main/scala/com/tecsisa/lightql/mat/elastic/Materializer.scala (hunk 1 above)
Comment (sergiohgz) [reply lightql-elastic/src/main/scala/com/tecsisa/lightql/mat/elastic/Materializer.scala]: Not at all identical, has a basic difference:
```scala
case EqOp.`=` => qb.must ...
case EqOp.!=  => qb.mustNot ...
```

**Review by juanjovazquez — APPROVED** [review]

**Pull request merged**